### PR TITLE
AMI Resolver

### PR DESCRIPTION
**Pull request by herebebogans**
Initial basic work on an AMI resolver. See README.md for syntax.

Sorry I'm not familiar with creating a spec for this.

### CODE REVIEW

File: lib/bora/resolver/ami.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+              values: [ami_prefix]
+            },
+            {
+              name:   'state',
+              values: ['available']
+            }
+          ]
+        ).images
+        raise NoAMI, "No Matching AMI's for prefix #{ami_prefix}" if images.empty?
+        images.sort! { |a, b| a.creation_date <=> b.creation_date }.last
```
Comment (ampedandwired) [inline lib/bora/resolver/ami.rb]: I think you need to return the `image_id` attribute from the image here:
`images.sort! { |a, b| a.creation_date <=> b.creation_date }.last.image_id

File: lib/bora/resolver/ami.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+require 'bora/cfn/stack'
+require 'pp'
```
Comment (ampedandwired) [inline lib/bora/resolver/ami.rb]: Is this needed?

**Comment by ampedandwired** [comment]
Thanks! Looks good except for it's returning the image object itself instead of the image_id attribute.

Yeah, the specs are a bit of a mess and hard to follow. I'll take care of it.

**Comment by herebebogans** [comment]
Updated.

**Comment by herebebogans** [comment]
I'd like to extend this in future with other filters describe_images can use eg Tag Names & Values but I'd thought start with something simple.

**Comment by ampedandwired** [comment]
Looks good, thanks for contributing. I think this implementation covers the most common use case anyway.
